lib/datadog/tracing/contrib/elasticsearch/patcher.rb

```diff
@@ -126,6 +127,25 @@ def datadog_configuration
           # rubocop:enable Metrics/MethodLength
           # rubocop:enable Metrics/AbcSize
 
+          # Patch to support both `elasticsearch` and `elastic-transport` versions
```

Comment:
@TonyCTHsu Is this the pattern we want to follow in situations where the instance we're trying to instrument may be different between versions of an integration? I didn't see any other direct examples of this (We do [patch pinning in mongodb](https://github.com/DataDog/dd-trace-rb/blob/master/lib/datadog/tracing/contrib/mongodb/instrumentation.rb#L20) for different reasons). 

I have two questions/concerns:
1. Is the complexity / indirection we're adding to our code worth the tradeoff of simpler configuration?
2. Are we potentially confusing users by masking ES implementation details like this? `client.transport` refers to a different object depending on ES version and we're providing a way to configure on an instance of that object. I could see the merit of an argument that we shouldn't be changing that object behind the scenes. 

I lean slightly towards not doing this, but I don't have a strong opinion. At the end of the day either the library or the user has to set a different instance depending on ES version. 

CC: @delner @marcotc in case you have any thoughts on a general strategy for these types of situations. We should probably agree on a pattern so we are consistent the next time it occurs. [Some context](https://github.com/DataDog/dd-trace-rb/pull/3399#discussion_r1468189941).

Reply:
Closing the loop on this: Spoke to @marcotc and @TonyCTHsu and we all agreed with this pattern.